[Docs] Add MLRun install does not include KFP by default (#7193)
mlrun/mlrun: docs/runtimes/images.md

The images are:

- `mlrun/mlrun`: An MLRun image includes preinstalled OpenMPI and other ML packages. Useful as a base image for simple jobs. 

> - `mlrun/mlrun-kfp`: The same as `mlrun/mlrun` with the addition of KFP 1.8 already installed. **This image includes KFP.**

Reviewer: it is quiet the same. not exactly. `mlrun/mlrun`, comes with pydantic v2 while mlrun-kfp comes with pydantic v1 oob. the gap is because kfp 1.8 dependency comes with more restrictions.
perhaps we can refine the text to say
```suggestion
- `mlrun/mlrun-kfp`: MLRun based image with KFP 1.8 preinstalled. **This image includes KFP.** and being used for compiling kfp workflows, for remote workflows.
```

- `mlrun/mlrun-gpu`: The same as `mlrun/mlrun` but for GPUs, including Open MPI. 
- `mlrun/ml-base`: Image for file acquisition, compression, Dask jobs, simple training jobs and other utilities.

> - `mlrun/jupyter`: An image with Jupyter giving a playground to use MLRun in the open source. Built on top of jupyter/scipy-notebook, with the addition of MLRun and several demos and examples. **This image includes KFP.**

Reviewer 2: This image doesn't actually include KFP, I was mistaken.

> - `mlrun/mlrun-api`: The server's image.

Reviewer: ```suggestion
- `mlrun/mlrun-api`: The server's backend image.
```

> - `mlrun/log-collector`: A service that collects the logs for all runs and stores them in a persistent storage

Reviewer: ```suggestion
- `mlrun/log-collector`: A service that collects the logs for all runs and stores them in a persistent storage
- `mlrun/mlrun-ui`: The server's frontend image
```

```{admonition} Notes
When using the `mlrun` or `mlrun-gpu` image, use PyTorch versions up to and including than 2.0.1, but not higher. 
You can build your own images with newer CUDA for later releases of PyTorch.
```
### Install with or without KFP?

> MLRun uses KFP 1.8. If you want to use packages whose dependencies conflict with KFP 1.8, such as pydantic 2, then use any image except `mlrun/mlrun-kfp`. You can then install the package in your environment and your images without affecting the MLRun functionality.  

Reviewer: ```suggestion
MLRun uses KFP 1.8. If you want to use packages whose dependencies conflict with KFP 1.8, such as pydantic 2, then use any image except `mlrun/mlrun-kfp`. You can then install the package in your environment and your images without affecting the MLRun functionality.  
```

`mlrun-kfp` image is used to compile the workflow (& perform some mlrun agent-y operations like sending notifications, etc). the workflow functions may run based on `mlrun/mlrun` image which comes with pydantic v2.

Author: @liranbg Pls give a few more details. You compile a workflow with one image, and the functions use another image? Can you give an example?

- `mlrun/mlrun-kfp`: The same as `mlrun/mlrun` with the addition of KFP 1.8 already installed. Use it for compiling KFP workflows, and for remote workflows. **This image includes KFP.**
- `mlrun/mlrun-gpu`: The same as `mlrun/mlrun` but for GPUs, including Open MPI. 
- `mlrun/ml-base`: Image for file acquisition, compression, Dask jobs, simple training jobs and other utilities.
- `mlrun/jupyter`: An image with Jupyter giving a playground to use MLRun in the open source. Built on top of jupyter/scipy-notebook, with the addition of MLRun and several demos and examples. 
- `mlrun/mlrun-api`: The server's backend image.
- `mlrun/log-collector`: A service that collects the logs for all runs and stores them in a persistent storage
- `mlrun/mlrun-ui`: The server's frontend image

```{admonition} Notes
- When using the `mlrun` or `mlrun-gpu` image, use PyTorch versions up to and including than 2.0.1, but not higher. You can build your own images with newer CUDA for later releases of PyTorch.
-  If you are using a zipped source, use `mlrun/mlrun` images or install `unzip` in the provided function base image. 
```
### When to use an image with KFP

> MLRun supports KFP SDK 1.8. In general, if your workflow will be compiled locally,  meaning you are not working with a remote source, then you need to use the image `mlrun/mlrun-kfp`. If your workflow will be compiled remotely, or directly in KFP, then you can use any of `mlrun/mlrun`.

Reviewer 2: If `kfp` is installed locally, the user can compile their workflow locally (using the `kfp` engine).
Otherwise, the user can choose to use the `remote` engine, which will result in the workflow being compiled on the server side, using the `workflow-runner` pod, which uses the `mlrun-kfp image`.

Author: @adamdelman Isn't the starting point how you work - which determines the image you need? Not if you do or don't install kfp locally?

Reviewer 2: If you are using the `remote` engine, the workflow runner is used and automatically selects the `mlrun-kfp` image if not otherwise specified.
If KFP is not installed, you can only use this engine type to run workflows.
Unless you are using KFP specific code inside the MLRun job, you do not need to use the `mlrun-kfp` image directly.
